test/DocumentFormat.OpenXml.Packaging.Tests/OpenXmlPackageTests.cs

```diff
@@ -301,5 +304,31 @@ public void TestOpenModel3DWrittenByPowerPoint_DashMime()
 
             Assert.Equal("model/gltf-binary", model3DReferenceRelationshipPart.ContentType);
         }
+
+        // Test issue #1281 for regressions.
+        // When opening a workbook (SpreadsheetDocument.Open) with a missing calcChain part and default OpenSettings
+        // specified, we should throw an exception.
+        [Fact]
+        public void OpenWithMissingCalcChainPart()
+        {
+            Stream stmSpd = GetStream(TestFiles.MissingCalcChainPart, false);
+            Action act = () => SpreadsheetDocument.Open(stmSpd, false);
+            InvalidOperationException exception = Assert.Throws<InvalidOperationException>(act);
+
+            Assert.Equal("Specified part does not exist in the package.", exception.Message);
```

Comment:
Should never assert of exception messages - if you need to, use the resource strings

Reply:
or maybe just assert not null on the exception var? I really just want to verify the default behavior is to throw some exception for that package. I was hoping to be more specific but would it be acceptable to just assert that it throws?

Reply:
proposed code added